### PR TITLE
Better OAuth stuff thanks to Jenny's feedback

### PR DESCRIPTION
**Pull request by maelle**
Cf review https://github.com/rladies/meetupr/pull/84#pullrequestreview-576887302 Thanks a ton @jennybc, this was most useful. :pray: 

@beatrizmilz @jeanprado @benubah This will in particular change where the token is cached by default (an [app dir](https://blog.r-hub.io/2020/03/12/user-preferences/#not-so-temporary-files3)) and how you can indicate to meetupr where to find an existing token (no longer via an environment variable, but instead via the `token` argument of `meetup_auth()`).

Sorry to change things again but the previous iteration was dangerous. We'd better get this right before the CRAN release. :grimacing:

### CODE REVIEW

File: R/get_pro.R [hunk 1]
```diff
@@ -92,7 +92,7 @@ get_pro_events <- function(urlname,
   api_path <- sprintf("pro/%s/events", urlname)
   res <- .fetch_results(api_path = api_path, verbose = verbose)
 
-  group <- lapply(res, function(x) x[[3]])
+  group <- lapply(res, function(x) x[["chapter"]])
```
Comment (maelle) [inline R/get_pro.R:95]: @Athanasiamo sometimes x is of length 2 only, when the event has no image (or the chapter had no image? I can't remember now)

**Review by ledell — APPROVED** [review]
This looks great to me, thank you @maelle and @jennybc!

File: R/get_pro.R (hunk 1 above)
Comment (drmowinckels) [reply R/get_pro.R:95]: great! ys, definitely better to index by name!

**Review by drmowinckels — APPROVED** [review]

**Comment by maelle** [comment]
Thanks @ledell @Athanasiamo :pray:

**Pull request merged**